qiskit/providers/backend_compat.py

```diff
@@ -13,24 +13,367 @@
 """Backend abstract interface for providers."""
 
 from __future__ import annotations
-
-from typing import List, Iterable, Any, Dict, Optional
+import logging
+from typing import List, Iterable, Any, Dict, Optional, Union, Tuple
 
 from qiskit.exceptions import QiskitError
 
 from qiskit.providers.backend import BackendV1, BackendV2
 from qiskit.providers.backend import QubitProperties
 from qiskit.utils.units import apply_prefix
-from qiskit.circuit.library.standard_gates import get_standard_gate_name_mapping
-from qiskit.circuit.measure import Measure
+from qiskit.providers.models.backendconfiguration import (
+    QasmBackendConfiguration,
+    PulseBackendConfiguration,
+)
+
 from qiskit.providers.models.backendconfiguration import BackendConfiguration
 from qiskit.providers.models.backendproperties import BackendProperties
+
 from qiskit.providers.models.pulsedefaults import PulseDefaults
 from qiskit.providers.options import Options
 from qiskit.providers.exceptions import BackendPropertyError
+from qiskit.providers.models.backendproperties import Gate as GateSchema
+
+
+logger = logging.getLogger(__name__)
+
+
+class IBMQubitProperties(QubitProperties):
+    """A representation of the properties of a qubit on an IBM backend."""
+
+    __slots__ = (  # pylint: disable=redefined-slots-in-subclass
+        "t1",
+        "t2",
+        "frequency",
+        "anharmonicity",
+        "operational",
+    )
+
+    def __init__(  # type: ignore[no-untyped-def]
+        self,
+        t1=None,
+        t2=None,
+        frequency=None,
+        anharmonicity=None,
+        operational=True,
+    ):
+        """Create a new ``IBMQubitProperties`` object
+
+        Args:
+            t1: The T1 time for a qubit in secs
+            t2: The T2 time for a qubit in secs
+            frequency: The frequency of a qubit in Hz
+            anharmonicity: The anharmonicity of a qubit in Hz
+            operational: A boolean value representing if this qubit is operational.
+        """
+        super().__init__(t1=t1, t2=t2, frequency=frequency)
+        self.anharmonicity = anharmonicity
+        self.operational = operational
+
+    def __repr__(self):  # type: ignore[no-untyped-def]
+        return (
+            f"IBMQubitProperties(t1={self.t1}, t2={self.t2}, frequency={self.frequency}, "
+            f"anharmonicity={self.anharmonicity})"
+        )
```

Comment:
```suggestion
```
In this PR, Using `QubitProperties` is sufficient, I think. If we need it, we will figure it out.

Reply:
Applied suggestion :)

Reply:
Regarding the `IBMQubitProperties` I suspected if this class is even needed here, because we need to separate the IBM specific codes from qiskit. but since it was there already so I decided to changed the least and kept it as it is :)